exercises/practice/propositional-logic/propositional_logic_test.py

```diff
@@ -0,0 +1,54 @@
+import unittest
+from z3 import *
+from propositional_logic import *
+
+
+class PropositionalLogicTest(unittest.TestCase):
+    def test_A_B(self):
+        A, B = Bools('A B')
+        t1, t2, t3 = propositional_logic_proofs(A, B)
+
+        expected_t1 = And(A, B) == Not(Or(Not(A), Not(B)))
+        expected_t2 = Implies(Implies(A, Implies(A, B)), Implies(A, B))
+        expected_t3 = Implies(And(A, Implies(Not(B), Not(A))), B)
+
+        self.assertEqual(t1, expected_t1)
+        self.assertEqual(t2, expected_t2)
+        self.assertEqual(t3, expected_t3)
+
+    def test_notA_B(self):
+        A, B = Bools('A B')
+        t1, t2, t3 = propositional_logic_proofs(Not(A), B)
+        expected_t1 = And(Not(A), B) == Not(Or(Not(Not(A)), Not(B)))
+        expected_t2 = Implies(Implies(Not(A), Implies(Not(A), B)), Implies(Not(A), B))
+        expected_t3 = Implies(And(Not(A), Implies(Not(B), Not(Not(A)))), B)
+
+        self.assertEqual(t1, expected_t1)
+        self.assertEqual(t2, expected_t2)
+        self.assertEqual(t3, expected_t3)
+
+    def test_A_notB(self):
+        A, B = Bools('A B')
+        t1, t2, t3 = propositional_logic_proofs(A, Not(B))
+        expected_t1 = And(A, Not(B)) == Not(Or(Not(A), Not(Not(B))))
+        expected_t2 = Implies(Implies(A, Implies(A, Not(B))), Implies(A, Not(B)))
+        expected_t3 = Implies(And(A, Implies(Not(Not(B)), Not(A))), Not(B))
+
+        self.assertEqual(t1, expected_t1)
+        self.assertEqual(t2, expected_t2)
+        self.assertEqual(t3, expected_t3)
+
+    def test_notA_notB(self):
+        A, B = Bools('A B')
+        t1, t2, t3 = propositional_logic_proofs(Not(A), Not(B))
+        expected_t1 = And(Not(A), Not(B)) == Not(Or(Not(Not(A)), Not(Not(B))))
+        expected_t2 = Implies(Implies(Not(A), Implies(Not(A), Not(B))), Implies(Not(A), Not(B)))
+        expected_t3 = Implies(And(Not(A), Implies(Not(Not(B)), Not(Not(A)))), Not(B))
+
+        self.assertEqual(t1, expected_t1)
+        self.assertEqual(t2, expected_t2)
+        self.assertEqual(t3, expected_t3)
+
+
+if __name__ == "__main__":
+    unittest.main()
```

Comment:
Are there anymore tests that you can run (to get more than only 4 tests)? Or no because the properties don't apply to more?

Reply:
I think only those 4 can be run because they take on all the possible boolean value combinations of A and B

Reply:
Also @JimmyQuenichet You're going to have to mess with git and force push some. The commits on a pull request will have different hashes so your main and the exercism main will diverge when merged.